src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -291,17 +291,14 @@ private BoundExpression GetSpeculativelyBoundExpression(int position, Expression
                     var crefWrapper = SyntaxFactory.QualifiedCref(qualified.Left, SyntaxFactory.NameMemberCref(qualified.Right));
                     crefSymbols = BindCref(crefWrapper, binder);
                 }
-                else
+                else if (expression is TypeSyntax typeSyntax)
                 {
-                    var typeSyntax = expression as TypeSyntax;
-                    if (typeSyntax != null)
-                    {
-                        var crefWrapper = typeSyntax is PredefinedTypeSyntax ?
-                            (CrefSyntax)SyntaxFactory.TypeCref(typeSyntax) :
-                            SyntaxFactory.NameMemberCref(typeSyntax);
-                        crefSymbols = BindCref(crefWrapper, binder);
-                    }
+                    var crefWrapper = typeSyntax is PredefinedTypeSyntax ?
```

Comment:
Sure. should I consider this for like *all* type checks? there's a plenty in this file but it seems like that none of them are hot paths.